Documentation/README.md

```diff
@@ -57,3 +57,146 @@ tar -xf <tarball-destination> -C <path-to-extraction-root>
+
+```console
+build.{cmd|sh} /p:RootRepo=core-setup
+```
+
+Note that since you've changed the coreclr and corefx commits, the build will find the current commit changed so if you don't want to see messages like:
+```console
+WARNING: submodule src/corefx, currently at 831264e53e5b9333850baa659af8a2857a9cb9b7, has diverged from checked-in
+version 5b7674e4ae5cc782e99f50b2919dfdeb29106a46
+If you are changing a submodule branch or moving a submodule backwards, this is expected.
+Should I checkout src/corefx to the expected commit 5b7674e4ae5cc782e99f50b2919dfdeb29106a46 [N]o / [y]es / [q]uit
+```
+
+You can set `SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1` environment variable or after you've pulled the changes, before you build, in your `source-build` repo just add the changes with, `git add src/corefx` and/or `git add src/coreclr`.
+
+Once the build is done, we will have a .NET Runtime containing the `Hashtable.PNSE()` API that can be used with our local cli in .NET Core projects.
+
+### How to use the produced Runtime
+
+The produced runtime will be under `bin/obj/{architecture}/{configuration}/blob-feed/packages`.
+
+Now create a new .NET Core project:
+```console
+dotnet new Console -o source-build-test
+```
+
+In your new project add a new NuGet.config to point to the Microsoft.NETCore.App and runtime packages.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <!-- NOTE: Leave this file here and keep it in sync with list in dir.props. -->
+  <!-- The command-line doesn't need it, but the IDE does.                    -->
+  <packageSources>
+    <clear/>
+    <add key="local feed" value="C:\repos\source-build\bin\obj\x64\Release\blob-feed\packages" />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+  </packageSources>
+</configuration>
+```
+
+Then in your csproj we need to set the Runtime version:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
+    <RootNamespace>source_build_test</RootNamespace>
+    <RuntimeFrameworkVersion>2.2.0-preview1-26509-04</RuntimeFrameworkVersion>
+    <NETCoreAppMaximumVersion>2.2</NETCoreAppMaximumVersion>
+  </PropertyGroup>
+
+</Project>
+```
+
+Note that the `TargetFramework` value needs to be the minimum or later framework that the Runtime supports. Therefore if there is no cli produced yet or you don't have installed a dotnet cli that supports that `TargetFramework` you need to add the property `NETCoreAppMaximumVersion` to match the version framework you want to target.
+
+If you don't set this property, you will see an error similar to this one:
+```console
+C:\Program Files\dotnet\sdk\2.1.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(137,5): error : The current .NET SDK does not support targeting .NET Core 2.2.  Either target .NET Core 2.1 or lower, or use a version of the .NET SDK that supports .NET Core 2.2.
```

Comment:
I tend to use quotes rather than code blocks for error messages because I hate scrolling to read long lines. 😃